### PR TITLE
trivial: Untangle FuHistory from FuPlugin

### PR DESCRIPTION
**Pull request by hughsie**
This will allow us to export FuPlugin in the future without dragging in the
private fu-history.h too.

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -1613,6 +1614,144 @@ fu_engine_is_running_offline (FuEngine *self)
...
+
+	/* create symlink for the systemd-system-update-generator */
+	rc = symlink (symlink_target, trigger);
+	if (rc < 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INTERNAL,
+			     "Failed to create symlink %s to %s: %s",
+			     trigger,
+			     "/var/lib", strerror (errno));
```
Comment (gicmo) [inline src/fu-engine.c]: There is `g_strerror`, which compared to `strerror` is thread-safe. Not sure it is important though.

Comment (superm1) [reply src/fu-engine.c]: This line shouldn't hardcode /var/lib, should look up string to use, no?

File: src/fu-engine.c [hunk 2]
```diff
@@ -1613,6 +1614,144 @@ fu_engine_is_running_offline (FuEngine *self)
...
+ * @release: A #FwupdRelease
+ * @blob_cab: A #GBytes
+ * @flags: #FwupdInstallFlags
+ * @error: A #GError or NULL
+ *
+ * Schedule an offline update for the device
+ *
+ * Returns: #TRUE for success, #FALSE for failure
+ *
+ * Since: 1.3.6
```
Comment (superm1) [inline src/fu-engine.c]: Planning to skip 1.3.5?

Comment (hughsie) [reply src/fu-engine.c]: No, just need coffee

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c]: > There is g_strerror

good catch, thanks.

Comment (hughsie) [reply src/fu-engine.c]: > This line shouldn't hardcode /var/lib

I don't think it matters what the destination is, although I guess it makes sense to chose something sensible.

Comment (hughsie) [reply src/fu-engine.c]: Hah, I already did that, the error message was just wrong. Fixed.

**Pull request merged**